File: src/main/java/adapter/repository/ExchangeRateV2Repository.java
package adapter.repository;

import adapter.Currency;
import adapter.ExchangeRateV2;

public class ExchangeRateV2Repository implements ExchangeRateV2 {
    @Override
    public double getExchangeRate(Currency from, Currency to) {
        return switch (from) {
            case USD -> fromUSD(to);
            case KRW -> fromKRW(to);
        };
    }

    private double fromUSD(Currency to) {
        return switch (to) {
            case KRW -> 1271.14;
            case USD -> 1;
        };
    }

    private double fromKRW(Currency to) {
        return switch (to) {
            case USD -> 0.00079;
            case KRW -> 1;
        };
    }
}

File: src/main/kotlin/adapter/repository/ExchangeRateV2Repository.kt
package adapter.repository

import adapter.Currency
import adapter.ExchangeRateV2

class ExchangeRateV2Repository : ExchangeRateV2 {
    override fun getExchangeRate(from: Currency, to: Currency): Double {
        return when (from) {
            Currency.USD -> fromUSD(to)
            Currency.KRW -> fromKRW(to)
        }
    }

    private fun fromUSD(to: Currency): Double {
        return when (to) {
            Currency.KRW -> 1271.14
            Currency.USD -> 1.0
        }
    }

    private fun fromKRW(to: Currency): Double {
        return when (to) {
            Currency.USD -> 0.00079
            Currency.KRW -> 1.0
        }
    }
}